volatility3/framework/plugins/linux/capabilities.py

```diff
@@ -0,0 +1,228 @@
+
+        # Ambient capabilities were added in kernels 4.3.6
+        if task_cred.has_member("cap_ambient"):
+            fields["capabilities"].append(task_cred.cap_ambient)
+        else:
+            fields["capabilities"].append(renderers.NotAvailableValue())
+
+        return fields
+
+    def get_tasks_capabilities(
+        self, tasks: List[interfaces.objects.ObjectInterface]
+    ) -> Iterable[Dict]:
+        """Yields a dict for each task containing the task's basic information along with its capabilities
+
+        Args:
+            tasks: An iterable with the tasks to process.
+
+        Yields:
+            Iterable[Dict]: A dict for each task containing the task's basic information along with its capabilities
+        """
+        for task in tasks:
+            if task.is_kernel_thread:
+                continue
+
+            yield self.get_task_capabilities(task)
+
+    def _generator(
+        self, tasks: Iterable[interfaces.objects.ObjectInterface]
+    ) -> Iterable[Tuple[int, Tuple]]:
+        for fields in self.get_tasks_capabilities(tasks):
+            selected_fields = fields["common"]
+            cap_inh, cap_prm, cap_eff, cap_bnd, cap_amb = fields["capabilities"]
+
+            if self.config.get("inheritable"):
+                selected_fields.append(self._decode_cap(cap_inh))
+            elif self.config.get("permitted"):
+                selected_fields.append(self._decode_cap(cap_prm))
+            elif self.config.get("effective"):
+                selected_fields.append(self._decode_cap(cap_eff))
+            elif self.config.get("bounding"):
+                selected_fields.append(self._decode_cap(cap_bnd))
+            elif self.config.get("ambient"):
+                selected_fields.append(self._decode_cap(cap_amb))
+            else:
+                # Raw values
+                selected_fields.append(format_hints.Hex(cap_inh.get_capabilities()))
+                selected_fields.append(format_hints.Hex(cap_prm.get_capabilities()))
+                selected_fields.append(format_hints.Hex(cap_eff.get_capabilities()))
+                selected_fields.append(format_hints.Hex(cap_bnd.get_capabilities()))
+
+                # Ambient capabilities were added in kernels 4.3.6
+                if isinstance(cap_amb, renderers.NotAvailableValue):
+                    selected_fields.append(cap_amb)
+                else:
+                    selected_fields.append(format_hints.Hex(cap_amb.get_capabilities()))
+
+            yield 0, selected_fields
+
+    def run(self):
+        pids = self.config.get("pids")
+        pid_filter = pslist.PsList.create_pid_filter(pids)
+        tasks = pslist.PsList.list_tasks(
+            self.context, self.config["kernel"], filter_func=pid_filter
+        )
+
+        columns = [
+            ("Name", str),
+            ("Tid", int),
+            ("Pid", int),
+            ("PPid", int),
+            ("EUID", int),
+        ]
+
+        if self.config.get("inheritable"):
```

Comment:
Generally altering columns and/or conditionals around the yield step of the generator are what start me looking closer.  I think there's a couple of very specific places I've let it in, but this is a lot of fields to do it with, so I'd prefer to get this reworked...